src/mqt/qecc/ft_stateprep/state_prep_det.py

```diff
@@ -0,0 +1,811 @@
+"""Synthesizing deterministic state preparation circuits for d<5 CSS codes."""
+
+from __future__ import annotations
+
+import logging
+from itertools import product
+from typing import TYPE_CHECKING
+
+import numpy as np
+import z3
+from ldpc import mod2
+
+from .state_prep import (
+    StatePrepCircuit,
+    _coset_leader,
```

Comment:
In https://github.com/cda-tum/mqt-qecc/pull/275, a lot of stuff has already been refactored since I needed to reuse some of the stateprep methods anyway. It will be a bit before that one is merged, though. I am fine with making `__coset_leader` public for now since it is the easiest solution for @lsschmid right now.

Reply:
Note that it is not only this function alone that is being imported, see
https://github.com/cda-tum/mqt-qecc/blob/acd120a3b39d6e24074019ee762a89a024cbd429/src/mqt/qecc/ft_stateprep/state_prep_det.py#L15-L20

Reply:
All of these are refactored and made public in the mentioned PR. Making them public in this PR is therefore compatible with my changes.

Reply:
Nice. Then let's go with that here.

Reply:
I think it makes sense to keep state_prep and state_prep_det in separate files as the code is mostly unrelated to each other. Only point are these helper functions which Tom plans to move to a separate file anyway, so we can do the restructuring there.

Reply:
Just to be on the safe side: the functions should still be modified to be public (removing the leading underscores) in this PR here before the whole code can be restructured in a follow-up PR.